sql/smoot_usage_v1.sql

```diff
@@ -10,7 +10,17 @@ WITH
       'Firefox Desktop Dev Tools Opened',
       NULL) ] AS usage_ids
   FROM
-    `moz-fx-data-derived-datasets.telemetry.clients_daily_v6` )
+    `moz-fx-data-derived-datasets.telemetry.smoot_clients_daily_1percent_v1` ),
+  --
+  usage_arrays_7_day_delay AS (
```

Comment:
The major new concept here is that I believe we'll be able to support all the metrics that depend on activity in a forward-looking window directly in clients_last_seen rather than having to create additional tables.

We are tracking "days_since_created_profile" as part of creating each new day of clients_last_seen. When this value is equal to 6, it means that the user should count for the "New Firefox Desktop Profile Created" usage criterion _7 days ago_. The CTE `usage_arrays_7_day_delay` is a way of essentially sliding backwards 7 days so that clients_last_seen represents a window not of the previous 28 days, but of the previous 21 and next 7.

In the live view created in this file, we union the 7 day delayed query with the non-delayed query to get all usage criteria aligned to the correct day. You'll see no rows with usage_id "New Firefox Desktop Profile Created" if you're looking at recent days.

I think we can adapt this to our exact_mau datasets by having a separate view for "1 week metrics" on top of the raw exact_mau table. The date presented in the view will be the 7 day delayed version of submission_date.

By adding in our idea of maintaining arrays of usage over the past 28 days (rather than or in addition to days_since_*; see ), I believe this technique will allow us present 1-week, 2-week, and 3-week retention metrics as views within our exact_mau28 methodology.